src/hdmf_zarr/backend.py

```diff
@@ -344,7 +359,8 @@ def write_group(self, **kwargs):
                 self.write_dataset(parent=group,
                                    builder=sub_builder,
                                    link_data=link_data,
-                                   exhaust_dci=exhaust_dci)
+                                   exhaust_dci=exhaust_dci,
+                                   export_source=export_source)
```

Comment:
@mavaylon1 references are also stored in attributes. We will need to pass `export_source` also to `write_attributes` so that it can pass the `export_source` also in its calls to `__get_ref`

Reply:
You may want to search for `__get_ref` to make sure that all places where `__get_ref` is used on write are passed the `export_source` parameter.

Reply:
Ah missed those. Good catch